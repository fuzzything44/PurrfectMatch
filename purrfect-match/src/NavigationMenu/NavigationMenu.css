.Header {
    position: fixed;
    width: 100%;
    z-index: 100;
    transition: 0.8s cubic-bezier(0.1, 0.8, 0.2, 1);
    right:0px;
    padding: 15px 0;
}

.HeaderScrolled {
    background:#9F63F9;
    padding: 15px 0;
    backdrop-filter: blur(20px);
}

.HeaderGroup {
    max-width: 800;
    margin: 0px auto;
    display: grid;
    grid-template-columns: repeat(5, auto);
    align-items: center;
    justify-items: center;
}

.Header a {
    color: black;
    font-weight: 300;
}

.Header button {
    padding: 5px 5px;
    font-size: 20px;
    background: rgba(255,255,255,0.0);
    border: none;
    font-weight: 700;
    border-radius: 5px;
    outline: none;
    cursor: pointer;
    transition: 1s cubic-bezier(0.2, 0.8, 0.2, 1)
}


.Header button:hover {
    background:  rgba(255,255,255,0.1);
    box-shadow: 0 10px 20px rgba(0, 0, 0, 0.25);
    transform: translateY(-3px);
}

@media (max-width: 640px) {
    .Header {
        padding: 15px 0;
    }

    .HeaderGroup {
        grid-template-columns: repeat(4, auto);
    }

    .Header a:nth-child(4) {
        display: none;
    }
}





