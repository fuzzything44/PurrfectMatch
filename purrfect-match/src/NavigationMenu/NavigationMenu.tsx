import * as React from "react";
import './NavigationMenu.css';
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import logo from "pictures/doglogo.png";


interface MenuProps {
    
  
};


interface MenuState{
  hasScrolled:boolean;

}




class NavigationMenu extends React.Component< MenuProps,MenuState> {
  constructor(props:any) {
    super(props)

    this.state = {
      hasScrolled: false
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', this.handleScroll)
  }

  handleScroll = (event:any) => {
    const scrollTop = window.pageYOffset
    this.setState({ hasScrolled: scrollTop > 50 });
  }

 

  render() {
    return (
      <nav className={this.state.hasScrolled ? 'Header HeaderScrolled' : 'Header'}>
        <div className="HeaderGroup ">
        
        <Link to="home"><img id="logo"src={logo} alt="Purrfect Match Logo"/></Link>
        
       
         
        <h1>PurrfectMatch </h1>
        <button>
        <Link to="success"> Success Stories</Link>
        </button>
        
        <button>
        <Link to="contact"> Contact Us</Link>
        </button>
        
        </div>
    
      </nav>
    );
  }
}

export default NavigationMenu