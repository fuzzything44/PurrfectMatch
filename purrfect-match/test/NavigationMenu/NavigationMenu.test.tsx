import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NavigationMenu } from 'NavigationMenu/NavigationMenu';

configure({ adapter: new Adapter() });

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
    wrapper = shallow(<NavigationMenu />);
});



test("Renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
    // Scroll
    wrapper.setState({ hasScrolled: true });
    expect(wrapper).toMatchSnapshot();
});

test("handleScroll updates correctly", () => {
    // This component uses the global "window" object to see the scroll of the page
    // Sadly, this isn't easy to work with as it's readonly, so we need to do a lot of setup
    // to be able to mock it. 
    let spy = jest.spyOn(window, 'addEventListener');
    let yVal = 0;
    Object.defineProperty(window, "pageYOffset", { get: () => yVal });

    // Mount the component. Since we shallow rendered this wasn't called.
    wrapper.instance().componentDidMount();
    let scrollCallback: any = spy.mock.calls[0][1]; // Second parameter passed in for first call to this function. Should be the scroll handler

    expect(scrollCallback).toBe(wrapper.instance().handleScroll);

    expect(wrapper.state("hasScrolled")).toBeFalsy();

    // Scroll to 50, 1 px away from changing
    yVal = 50;
    scrollCallback();
    expect(wrapper.state("hasScrolled")).toBeFalsy();

    // Scroll to 51
    yVal = 51;
    scrollCallback();
    expect(wrapper.state("hasScrolled")).toBeTruthy();

    // Scroll to 10
    yVal = 10;
    scrollCallback();
    expect(wrapper.state("hasScrolled")).toBeFalsy();
});
