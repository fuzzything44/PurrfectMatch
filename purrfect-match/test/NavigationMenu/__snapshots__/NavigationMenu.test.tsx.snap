// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<nav
  className="Header"
>
  <div
    className="HeaderGroup "
  >
    <Link
      to="/"
    >
      <div>
        <img
          alt="Purrfect Match Logo"
          id="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <p
          className="purrfectMatch"
        >
          Purrfect Match
        </p>
      </div>
    </Link>
    <button>
      <Link
        id="successStoriesLink"
        to="success"
      >
        Success Stories
      </Link>
    </button>
    <button>
      <Link
        id="contactLink"
        to="contact"
      >
        Contact Us
      </Link>
    </button>
  </div>
</nav>
`;

exports[`Renders correctly 2`] = `
<nav
  className="Header HeaderScrolled"
>
  <div
    className="HeaderGroup "
  >
    <Link
      to="/"
    >
      <div>
        <img
          alt="Purrfect Match Logo"
          id="logo"
          src="test-file-stub"
        />
      </div>
      <div>
        <p
          className="purrfectMatch"
        >
          Purrfect Match
        </p>
      </div>
    </Link>
    <button>
      <Link
        id="successStoriesLink"
        to="success"
      >
        Success Stories
      </Link>
    </button>
    <button>
      <Link
        id="contactLink"
        to="contact"
      >
        Contact Us
      </Link>
    </button>
  </div>
</nav>
`;
